crates/solver/src/settlement_access_list.rs

```diff
@@ -1,35 +1,59 @@
-use ethcontract::{Address, H256};
+use anyhow::{anyhow, Result};
+use ethcontract::{dyns::DynTransport, transaction::TransactionBuilder, Address, H256};
 use reqwest::{
     header::{HeaderMap, HeaderValue},
     Client, Url,
 };
 use serde::{Deserialize, Serialize};
-use web3::types::Bytes;
+use shared::Web3;
+use web3::types::{AccessList, Bytes};
+
+#[async_trait::async_trait]
+pub trait AccessListEstimating: Send + Sync {
+    async fn estimate_access_list(
+        &self,
+        web3: &Web3,
+        network_id: String,
+        tx: &TransactionBuilder<DynTransport>,
+    ) -> Result<AccessList> {
+        self.estimate_access_lists(web3, network_id, std::slice::from_ref(tx))
+            .await
+            .into_iter()
+            .next()
+            .unwrap()
+    }
+    async fn estimate_access_lists(
+        &self,
+        web3: &Web3,
+        network_id: String,
+        txs: &[TransactionBuilder<DynTransport>],
+    ) -> Vec<Result<AccessList>>;
```

Comment:
Do we need to have a batched operation? I can't think of any access list generation method that benefits from that.

I would propose just keeping `estimate_access_list` to keep it simpler.

Reply:
I thought it would be beneficial to batch several `eth_createAccessList` calls in a single RPC call? This is one place where we could benefit from it: https://github.com/gnosis/gp-v2-services/blob/7dd32c1290b4e9ff1a610e9feadf513ef0b2db70/crates/solver/src/settlement_simulation.rs#L36

Reply:
Do you think it's overkill?

Reply:
Ah good point. We can do a single batched Eth RPC call for the `eth_crateAccessList` implementation.

In that case I don't see why we shouldn't.